src/atd/yarn_v0.atd

```diff
@@ -0,0 +1,49 @@
+
+(**************************************************************************)
+(*  Copyright 2014, Sebastien Mondet <seb@mondet.org>                     *)
+(*                                                                        *)
+(*  Licensed under the Apache License, Version 2.0 (the "License");       *)
+(*  you may not use this file except in compliance with the License.      *)
+(*  You may obtain a copy of the License at                               *)
+(*                                                                        *)
+(*      http://www.apache.org/licenses/LICENSE-2.0                        *)
+(*                                                                        *)
+(*  Unless required by applicable law or agreed to in writing, software   *)
+(*  distributed under the License is distributed on an "AS IS" BASIS,     *)
+(*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or       *)
+(*  implied.  See the License for the specific language governing         *)
+(*  permissions and limitations under the License.                        *)
+(**************************************************************************)
+
+type time <ocaml from="Ketrew_gen_base_v0.Time"> = abstract
+type host <ocaml from="Ketrew_gen_base_v0.Host"> = abstract
+type path <ocaml from="Ketrew_gen_base_v0.Path"> = abstract
+type program <ocaml from="Ketrew_gen_base_v0.Program"> = abstract
+type monitored_script <ocaml from="Ketrew_gen_base_v0.Monitored_script"> = abstract
+
+type daemonized_run <ocaml from="Ketrew_gen_daemonize_v0.Running"> = abstract
+
+type distributed_shell_parameters = {
+  hadoop_bin: string;
+  distributed_shell_shell_jar: string;
+  container_memory: [ GB of int | Raw of string ];
+  timeout: [ Seconds of int | Raw of string ];
```

Comment:
> Leaving `Raw will just be confusing to the user. What's an example of something that you would experiment with? 

By experience with the ecosystem: 
Users are going to run into some crazy JVM problems, and somebody in some mailing-list is going to say that `-container-memory` is actually a brainfuck-like EDSL, and if you call it with `"10000:+3K~[42]" it is going to set some other parameter of the JVM's GC that is going to solve their particular problem (I'm making things up, but it sounds very realistic in the hadoop world).